packages/context/src/inject-config.ts

```diff
@@ -111,7 +140,8 @@ function resolveFromConfig(
   injection: Injection,
   session: ResolutionSession,
 ): ValueOrPromise<unknown> {
-  const bindingKey = getCurrentBindingKey(session);
+  const bindingKey =
+    injection.metadata.targetBindingKey || getCurrentBindingKey(session);
```

Comment:
This logic is repeated in three places. Let's extract a shared helper function.

E.g.

```ts
const bindingKey = getConfigInjectionKey(injection, session);
```